### PR TITLE
new syslog changes

### PR DESCRIPTION
**Pull request by MiguelMoll**
New behavior for using syslog service within a VPC.

Related: https://github.com/convox/rack/pull/1381

### CODE REVIEW

File: _services/syslog.md [hunk 1]
```diff
@@ -41,3 +41,12 @@ To delete the syslog forwarder, use the `convox services delete` command:
 
     $ convox services delete syslog-3785
     Deleting syslog-3785... DELETING
+
+## VPC Access
+
+There are use cases for the URL used by the syslog service to resolve to an internal process within
```
Comment (mattmanning) [inline _services/syslog.md]: Is this paragraph basically saying that everything works if you create the syslog service *after* making your rack private?

File: _services/syslog.md [hunk 2]
```diff
@@ -41,3 +41,12 @@ To delete the syslog forwarder, use the `convox services delete` command:
...
+a Rack. This can be accomplished by simply creating the service while running a Rack with [private networking](/docs/private-networking/).
+
+If a syslog service exist and a Rack's networking mode is changed to private, the existing syslog would have to be recreated *if* internal access is needed. Otherwise it can stay as with just external internet access.
```
Comment (mattmanning) [inline _services/syslog.md]: Can you give example scenarios of internal access vs external access? I'm a little confused about cases where each would be used.

File: _services/syslog.md [hunk 3]
```diff
@@ -41,3 +41,12 @@ To delete the syslog forwarder, use the `convox services delete` command:
...
+
+Due to a known limitation with AWS Lambda, some planning is involved with syslog services and disabling a Rack's private network. Before the switch to public networking can be made, the existing syslog service (if it was created in private mode) has to be deleted. After being deleted, a Lambda function could leave an elastic network interface (ENI) with attached private subnets behind. We are in contact with AWS to address this limitation, but in the mean time the ENI must be manually deleted from the AWS VPC console. The ENI ID varies, but it's description in the AWS Console will begin with `AWS Lambda VPC ENI`. Once the ENI is manually detached and deleted, it is safe to disable private networking.
```
Comment (mattmanning) [inline _services/syslog.md]: I think this is basically saying, if you have a private rack using syslog created in private mode and you want to switch it to public, you need to delete the syslog service manually and recreate it after the switch. You might also have to clean up an ENI and some private subnets. Is that right?

It's not totally clear to me what happens if you try to switch to public mode without removing the syslog. Does the switch fail? Or are there other bad side effects?

File: _services/syslog.md [hunk 4]
```diff
@@ -44,9 +44,10 @@ To delete the syslog forwarder, use the `convox services delete` command:
...
+WARNING: If you have any syslog services created while your Rack was in private mode and you want to switch your Rack back to public mode, you will need to do some manual cleanup. This is an unfortunate, known limitation of AWS Lambda.
+
+Before switching your rack to public networking mode, log into the AWS VPC console. You will need to manually remove an Elastic Network Interface (ENI). The ENI ID varies, but its description in the AWS onsole will begin with `AWS Lambda VPC ENI`. Once the ENI is manually detached and deleted, it is safe to disable private networking. If you need assistance with this process, please open a support ticket at [https://console.convox.com](https://console.convox.com).
```
Comment (MiguelMoll) [inline _services/syslog.md]: I'd like to add that the syslog service should be deleted (`convox services delete syslog-1234`) before manually deleting the ENI.

Comment (mattmanning) [inline _services/syslog.md]: 👍 Adding that.

**Comment by MiguelMoll** [comment]
lgtm 👍

File: _services/syslog.md [hunk 5]
```diff
@@ -41,3 +41,15 @@ To delete the syslog forwarder, use the `convox services delete` command:
 
     $ convox services delete syslog-3785
     Deleting syslog-3785... DELETING
+
+## VPC Access
+
+In most cases you will be shipping logs via the syslog service to an external drain like Papertrail or Logentries, for example. However, in some cases you might want to ship the logs back to a syslog drain running in your Rack. For this to work, you'll need to first switch your rack to [private networking mode](/docs/private-networking/) before creating the syslog service.
```
Comment (mwarkentin) [inline _services/syslog.md]: ", for example" seems unnecessary here.

**Pull request merged**